corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -1225,6 +1243,63 @@ def get_case_transactions_for_form(form_id, limit_to_cases):
             for trans in resultset:
                 yield trans
 
+    @staticmethod
+    def get_case_transactions_by_case_id(case_id, updated_xforms=None):
+        """
+        This fetches all the transactions required to rebuild the case along
+        with all the forms for those transactions.
+
+        For any forms that have been updated it replaces the old form
+        with the new one.
+
+        :param case_id: ID of case to rebuild
+        :param updated_xforms: list of forms that have been changed.
+        :return: list of ``CaseTransaction`` objects with their associated forms attached.
+        """
+
+        transactions = CaseAccessorSQL.get_transactions_for_case_rebuild(case_id)
+        return CaseAccessorSQL.fetch_case_transaction_forms(transactions, updated_xforms)
+
+    @staticmethod
+    def fetch_case_transaction_forms(transactions, updated_xforms=None):
+        """
+        Fetches the forms for a list of transactions
+
+        :param transactions: list of ``CaseTransaction`` objects:
+        :param updated_xforms: list of forms that have been changed.
+        :return: list of ``CaseTransaction`` objects with their associated forms attached.
```

Comment:
This function returns the same list that was passed in as an argument (after mutating the items in that list). Functions that mutate their arguments (and then return a value) can cause surprises that are hard to reason about. What do you think of having this function return nothing, and update the docstring to clearly indicate that it mutates the `transactions` argument?

Reply:
makes sense 2f2325b